### ISSUE TITLE
"Try it out" button losing resource

### ISSUE THREAD

**Issue by LRScudeletti** [fenomeno83/SwaggerGlobalization#4]
Hi.

When I'm using the Italian resource, the button "Try it out" changes to "Prova", when I click "Prova" the button changes to "Cancella", if I click the button again it changes to "Try it out", but it should change to "Prova".

The Italian feature is lost for both a "Prova" and a "Cancella" word.

Thanks.

**Comment by fenomeno83** [comment]
Add in translate.js under this code

                  $.initialize('button.try-out__btn.cancel', function () {
                      $(this).html(resource_globalization["Cancel"]);
                   });

this new code

            $(document).click(function () {
                var targ = $(event.target);
        
                if (targ.is(".btn.try-out__btn.cancel")) {
                    targ.text(resource_globalization["Cancel"]);
                }
        
                if (targ.is(".btn.try-out__btn") && !targ.hasClass("cancel")) {
                    targ.text(resource_globalization["TryItOut"]);
                }
        
            });

I'll deploy a new version with these fixes

**Comment by LRScudeletti** [comment]
Hi. 

It worked correctly, thanks for the quick responses. During testing, I detected that the RequestUrl resource is also getting lost.

Finally, I would like to ask a question about the word "* required" and the validation "Please correct the following validation errors and try again. - Required field is not provided", can they not be translated via resource?

greetings

**Comment by fenomeno83** [comment]
you can use this ugly approach, checking english text and replacing it
in this case, in translate.js

       $.initialize('div.validation-errors.errors-wrapper', function () {

        if ($(this).html().startsWith("Please correct the following validation errors and try again.")) {
            $(this).html($(this).html().replace("Please correct the following validation errors and try again.", resource_globalization["ValidationError"]));

            var p = $(this).find('ul > li');
            p.each(function () {             
                if ($(this).html().includes("Required field is not provided")) {
                    $(this).html($(this).html().replace("Required field is not provided", resource_globalization["RequiredField"]));
                }
            });
            
        }
    });

in en.js
 "ValidationError": "Please correct the following validation errors and try again.",
    "RequiredField": "Required field is not provided"

for example in it.js
"ValidationError": "Perfavore, correggi i seguenti errori di validazione e riprova di nuovo.",
    "RequiredField": "Il campo obbligatorio non &egrave; stato compilato"


Now I'll deploy a new version.
Replace new translate.js in your code and check en.js/it.js for new keys

Regarding "required" label, it is more complex, because is on css..but i would keep untranslated "required" because it is like a keyword so no need to translate

**Comment by LRScudeletti** [comment]
Hi.

The message worked just fine that way.

About the other changes you made to the project, they also worked, with the exception of the "RequestURL", I changed it to the code below to work:

`$(this).children().first().html(resource_globalization["RequestUrl"]);`

The ExampleValue, Schema and Schemas strings were duplicated in your resource_globalization.

I added the option below to my project, it would be interesting for you to include it in your project too:

```
$.initialize('.view-line-link.copy-to-clipboard', function () {
    $(this).attr("title", resource_globalization["CopyClipboard"]);
});
```

Resource: "CopyClipboard": "Copy to clipboard"

Now I'm trying to apply the resources to the Expand and Collapse buttons, but I'm not succeeding because they have children that also expand and collapse.

Thanks.
